tests/testhttp.nim

```diff
@@ -1,76 +1,15 @@
 import unittest, json, strutils
-import httputils, chronicles
+import httputils
 import ../json_rpc/[rpcserver, rpcclient]
 
-const
-  TestsCount = 100
-  BufferSize = 8192
-  BigHeaderSize = 8 * 1024 + 1
-  BigBodySize = 128 * 1024 + 1
-  HeadersMark = @[byte(0x0D), byte(0x0A), byte(0x0D), byte(0x0A)]
+const TestsCount = 100
 
-  Requests = [
-    "GET / HTTP/1.1\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 71\r\n" &
-      "Content-Type: text/html\r\n" &
-      "Connection: close\r\n" &
-      "\r\n" &
-      "{\"jsonrpc\":\"2.0\",\"method\":\"myProc\",\"params\":[\"abc\", [1, 2, 3]],\"id\":67}",
-    "BADHEADER\r\n\r\n",
-    "GET / HTTP/1.1\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Type: application/json\r\n" &
-      "Connection: close\r\n" &
-      "\r\n",
-    "PUT / HTTP/1.1\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 71\r\n" &
-      "Content-Type: text/html\r\n" &
-      "Connection: close\r\n" &
-      "\r\n" &
-      "{\"jsonrpc\":\"2.0\",\"method\":\"myProc\",\"params\":[\"abc\", [1, 2, 3]],\"id\":67}",
-    "DELETE / HTTP/1.1\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 71\r\n" &
-      "Content-Type: text/html\r\n" &
-      "Connection: close\r\n" &
-      "\r\n" &
-      "{\"jsonrpc\":\"2.0\",\"method\":\"myProc\",\"params\":[\"abc\", [1, 2, 3]],\"id\":67}",
-    "GET / HTTP/0.9\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 71\r\n" &
-      "Content-Type: application/json\r\n" &
-      "\r\n" &
-      "{\"jsonrpc\":\"2.0\",\"method\":\"myProc\",\"params\":[\"abc\", [1, 2, 3]],\"id\":67}",
-    "GET / HTTP/1.0\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 71\r\n" &
-      "Content-Type: application/json\r\n" &
-      "\r\n" &
-      "{\"jsonrpc\":\"2.0\",\"method\":\"myProc\",\"params\":[\"abc\", [1, 2, 3]],\"id\":67}",
-    "GET / HTTP/1.1\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 71\r\n" &
-      "Content-Type: application/json\r\n" &
-      "Connection: close\r\n" &
-      "\r\n" &
-      "{\"jsonrpc\":\"2.0\",\"method\":\"myProc\",\"params\":[\"abc\", [1, 2, 3]],\"id\":67}",
-    "GET / HTTP/1.1\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 49\r\n" &
-      "Content-Type: application/json\r\n" &
-      "Connection: close\r\n" &
-      "\r\n" &
-      "{\"jsonrpc\":\"2.0\",\"method\":\"noParamsProc\",\"id\":67}",
-    "GET / HTTP/1.1\r\n" &
-      "Host: status.im\r\n" &
-      "Content-Length: 137438953472\r\n" &
-      "Content-Type: application/json\r\n" &
-      "Connection: close\r\n" &
-      "\r\n" &
-      "{128 gb Content-Length}",
-  ]
```

Comment:
we're not testing HTTP Server + Client anymore, we just assume that's done in `nim-chronos`

Reply:
Tests featuring invalid JSON-RPC payloads will be still quite useful.

Reply:
4c58d864151a2849d21e0b22d5dfb71baf61bff6